src/elm/type.coffee

```diff
@@ -112,10 +112,13 @@ module.exports.ToRatio = class ToRatio extends Expression
 
   exec: (ctx) ->
     arg = @execArgs(ctx)
-    if arg? and typeof arg != 'undefined'
+    if arg?
       # Argument will be of form '<quantity>:<quantity>'
       try
-        [numerator, denominator] = arg.toString().split(':').map((quantity) -> parseQuantity(quantity))
+        # String will be split into the following format ["", "numerator", "denominator", ""]
+        splitRatioString = arg.toString().split(/^(\d+\.\d+?\s*\'\w+\'\s*):(\s*\d+\.\d+?\s*\'\w+\'?)/)
```

Comment:
Wow.  I guess I don't understand `split` because I totally did not expect this to work!  I mean... you're splitting on a regex that captures the whole thing...  I guess I need to look up `split`.  Anyway, I think `match` would be the more common thing to use for this, wouldn't it?  It at least provides similar results but doesn't confuse dummies like me:
![image](https://user-images.githubusercontent.com/2278253/55571208-0dbba400-56d3-11e9-8f7c-2d51be95560b.png)

Also note that the captured numerator has a trailing space in it and the captured denominator has a leading space.  This is because you put the `\s*` inside of the `( )`.

Reply:
I also noticed, however, that this doesn't support strings when the `.0` is left off.  Is that intentional?  It seems like we should also support `1 'mg' : 5 'dL'`.
![image](https://user-images.githubusercontent.com/2278253/55571444-8589ce80-56d3-11e9-968b-af3098f50824.png)

Reply:
Last, I think the spec allows for ratios with no units, but the regex does not:
![image](https://user-images.githubusercontent.com/2278253/55571580-bff36b80-56d3-11e9-9a84-df8996cf1bf1.png)

Reply:
Sorry for the super long comments... but bottom-line, I've reworked the regex and tested it and I think this version works: `/^(\d+(\.\d+)?\s*('\w+')?)\s*:\s*(\d+(\.\d+)?\s*('\w+')?)$/`

It uses a lot more parens (to group the optional bits right), so you end up having to access index 1 for the numerator and index 4 for the denominator.

![image](https://user-images.githubusercontent.com/2278253/55572136-f41b5c00-56d4-11e9-855f-296e65b2a6b0.png)

BTW -- I *love* regex101.com for writing and testing regular expressions.  If you haven't used it, you should give it a go!  Here's a link to the suggested regex above: https://regex101.com/r/WDjDuQ/2